Adds eventing rabbitmq GA announcement blog (#5135)
knative/docs: blog/docs/articles/eventing-rabbitmq-announcement.md

# Announcing Eventing RabbitMQ moving to GA

**Author: Gab Satchi, Senior Software Engineer @ VMware**

**Date: 2022-07-26**

We’re excited to announce that [Eventing RabbitMQ](https://github.com/knative-sandbox/eventing-rabbitmq) is now GA (General Availability). With this milestone, the APIs provided will remain stable in future iterations.
A quick start can be found [here](https://knative.dev/docs/eventing/broker/rabbitmq-broker/).

# Key Features

### Use any RabbitMQ instance

Eventing RabbitMQ works well with RabbitMQ clusters provisioned using the [cluster operator](https://github.com/rabbitmq/cluster-operator) but can also be configured to use RabbitMQ instances provisioned by other means.

> Both broker and source have a `rabbitmqClusterReference` which can either reference a `RabbitMQCluster` created using the cluster operator or a `Secret` containing the credentials to a RabbitMQ instance.

Reviewer: ```suggestion
Both Broker and Source have a `rabbitmqClusterReference` which can either reference a `RabbitMQCluster` created using the cluster operator or a `Secret` containing the credentials to a RabbitMQ instance.
```

A full guide can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/external-cluster).

> ### FIFO vs Throughput

Reviewer 2: Please don't use shortened terms like vs, via, etc. It makes the docs vague and difficult to understand.
Maybe something simpler that properly describes the feature, e.g. "Parallelism and event ordering"?

Eventing RabbitMQ, by default, operates in a first-in-first-out (FIFO) mode for events. However, there can be use cases where ordering isn't required and higher throughput may be preferred.

> Bother broker and source have a `parallelism` annotation (default 1) that can be configured to process more than 1 message in parallel. An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).

Reviewer: ```suggestion
Brokers and Sources have a `parallelism` annotation (default 1) that can be configured to process more than 1 message in parallel. An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).
```

> Bother broker and source have a `parallelism` annotation (default 1) that can be configured to process more than 1 message in parallel. An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).

Reviewer: ```suggestion
Brokers and sources have a `parallelism` annotation (default 1) that can be configured to process more than 1 message in parallel. An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).
```

> Eventing RabbitMQ, by default, operates in a first-in-first-out (FIFO) mode for events. However, there can be use cases where ordering isn't required and higher throughput may be preferred.
> Bother broker and source have a `parallelism` annotation (default 1) that can be configured to process more than 1 message in parallel. An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).

Reviewer 2: ```suggestion
RabbitMQ by default uses a first-in-first-out (FIFO) ordering mode for events. However, there can be use cases where ordering isn't required and higher throughput may be preferred.
Both the RabbitMQ Broker and Source contain a `parallelism` annotation that can be configured to process more than one message in parallel.

An example outlining both FIFO and high throughput modes can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/samples/trigger-customizations).
```

save discussion of default values etc for the main docs that are pointed to here

> ### Queue Type

Reviewer 2: ```suggestion
### Quorum queues
```
Try to keep the headings relevant, i.e. actually describing the feature rather than a general topic

> Quorum queues are a fault tolerant replicated queue type to be used in multi-node setups. The performance between quorum queues and classic queues are very close and Eventing RabbitMQ has opted to using quorum queues by default.

Reviewer: ```suggestion
Quorum Queues are a fault tolerant replicated queue type to be used in multi-node setups. The performance between Quorum Queues and Classic Queues are very close and Eventing RabbitMQ has opted to using Quorum Queues by default.
```

> The queue type remains configurable at the broker level and any triggers subscribed to a broker will inherit the queue type. Queue type can be configured using the `RabbitMQBrokerConfig` object:

Reviewer: ```suggestion
The Queue type remains configurable at the Broker level and any Triggers subscribed to a Broker will inherit the Queue type. Queue type can be configured using the `RabbitMQBrokerConfig` object:
```

> Quorum queues are a fault tolerant replicated queue type to be used in multi-node setups. The performance between quorum queues and classic queues are very close and Eventing RabbitMQ has opted to using quorum queues by default.
> The queue type remains configurable at the broker level and any triggers subscribed to a broker will inherit the queue type. Queue type can be configured using the `RabbitMQBrokerConfig` object:
> 
> ```yaml
> apiVersion: eventing.knative.dev/v1alpha1
> kind: RabbitmqBrokerConfig
> ...
> spec:
>   queueType: "classic" #'classic | quorum'
> ...
> ```

Reviewer 2: ```suggestion
Quorum queues are a fault tolerant, replicated queue type that are used in multi-node setups. RabbitMQ uses quorum queues by default.

The queue type can be configured for RabbitMQ Brokers, and any Triggers subscribed to a Broker will inherit the queue type. You can configure the queue type for a `RabbitMQBrokerConfig` object as follows:

```yaml
apiVersion: eventing.knative.dev/v1alpha1
kind: RabbitmqBrokerConfig
...
spec:
  queueType: "classic" #'classic | quorum'
...
```
```

Tbh I think this example should be contained in the main docs somewhere instead and linked to from here, rather than including specific YAML configs in a blog

### Metrics

> Eventing RabbitMQ mirrors a subset of the eventing metrics found [here](https://knative.dev/docs/eventing/observability/metrics/eventing-metrics/). Due to the single-tenant nature, `Broker - Filter` metrics aren't applicable.

Reviewer 2: In the repo it's titled as "RabbitMQ Eventing components", so I think a proper name for this needs to be decided on and used consistently. If the title isn't going to be updated in the sandbox repo, the docs should also refer to "RabbitMQ Eventing components" throughout, not "Eventing RabbitMQ"

Author: Updating the repo title and docs to have a consistent name for the project

### Architecture

> Eventing RabbitMQ follows a single tenant model where dedicated resources (pods) are created per Broker, Trigger and Source objects. While there is some overhead to this approach, there are also benefits:

Reviewer 2: ```suggestion
Eventing RabbitMQ follows a single tenant model where dedicated resources are created for each Broker, Trigger, and Source object. While there is some overhead to this approach, there are also benefits:
```

Reviewer 2: I don't think the parentheses / pods is needed here - either use resources or pods

> - Pods can be autoscaled to handle the expected traffic for a given Broker/Trigger/Source.

Reviewer 2: ```suggestion
- Pods can be autoscaled to handle the expected traffic for a given Broker, Trigger, or Source.
```

please don't use slashes like this

- User's existing policies and roles can be used for role based access control (RBAC).

> - Increased transparency into what's being run on the cluster.

Reviewer 2: ```suggestion
- Increased transparency into what is running on the cluster.
```

<img src="/blog/articles/images/eventing-rabbitmq-architecture-broker.png" alt="Eventing RabbitMQ broker architecture" width="960"/>

> Creating a broker will create an ingress pod that can receive CloudEvents and RabbitMQ resources needed to plumb the event through the RabbitMQ system. A trigger will create a dispatcher pod that will consume messages from RabbitMQ and send the event to an addressable resource.

Reviewer: ```suggestion
Creating a Broker will create an Ingress pod that can receive CloudEvents and RabbitMQ resources needed to plumb the event through the RabbitMQ system. A Trigger will create a Dispatcher pod that will consume messages from RabbitMQ and send the event to an addressable resource.
```

> Creating a broker will create an ingress pod that can receive CloudEvents and RabbitMQ resources needed to plumb the event through the RabbitMQ system. A trigger will create a dispatcher pod that will consume messages from RabbitMQ and send the event to an addressable resource.

Reviewer 2: ```suggestion
Creating a Broker automatically creates an ingress pod that can receive CloudEvents, and the RabbitMQ resources needed to route CloudEvents through the RabbitMQ system. Triggers create dispatcher pods that consume messages from RabbitMQ, and send CloudEvents to an addressable resource.
```

<img src="/blog/articles/images/eventing-rabbitmq-architecture-source.png" alt="Eventing RabbitMQ source architecture" width="841"/>

> A RabbitMQSource object can retrieve messages from a queue, convert it to a cloud event and send it to an addressable consumer.

Reviewer: ```suggestion
A RabbitMQSource object can retrieve messages from a Queue, convert it to a CloudEvent and send it to an addressable consumer.
```

> A RabbitMQSource object can retrieve messages from a queue, convert it to a cloud event and send it to an addressable consumer.

Reviewer 2: ```suggestion
RabbitMQSource objects can retrieve messages from a queue, convert them to CloudEvents, and send these CloudEvents to addressable consumers.
```

### Performance

A suite of performance tests are run with each release to ensure adequate throughput and acceptable latency. The results of the latest run can be found [here](https://github.com/knative-sandbox/eventing-rabbitmq/tree/49787466f88b21fe022216a13c8321b334f92dc4/test/performance/results/release-v1.6).

> ### Adopters and Feedback
> Knative Eventing RabbitMQ is currently in use by VMware Event Broker Appliance ([VEBA](https://flings.vmware.com/vmware-event-broker-appliance)), Cloud Native Runtimes ([CNR](https://docs.vmware.com/en/Cloud-Native-Runtimes-for-VMware-Tanzu/1.3/tanzu-cloud-native-runtimes/GUID-cnr-overview.html)) and others. We will appreciate community feedback on trying this new release and reporting back any issues or queries that you might have.

Reviewer 2: ```suggestion
### Adopters

Knative Eventing RabbitMQ is currently used by VMware Event Broker Appliance ([VEBA](https://flings.vmware.com/vmware-event-broker-appliance)) and Cloud Native Runtimes ([CNR](https://docs.vmware.com/en/Cloud-Native-Runtimes-for-VMware-Tanzu/1.3/tanzu-cloud-native-runtimes/GUID-cnr-overview.html)).
```

Remove the bit about feedback unless you're going to give instructions on how to send it. If so, please add a new heading for ### Feedback and add details there.
Also, as mentioned above, please decide on and use consistent naming for this feature throughout the blog - "Knative Eventing RabbitMQ" isn't used elsewhere, it's Eventing RabbitMQ in the doc and RabbitMQ Eventing components in the repo. Please make this consistent across the board.
